cloudsync_onedrive.py

```diff
@@ -817,10 +818,7 @@ def rename(self, oid, path):  # pylint: disable=too-many-locals, too-many-branch
             try:
                 updated = False
                 if info.name != base:
-                    need_temp = False
                     if info.name.lower() == base.lower():
-                        need_temp = True
-                    if need_temp:
```

Comment:
Can you provide some more information about why every rename is required to be a 2-step rename? 2-step renames are inherently problematic, because casual exceptions (such as rate limit errors) can happen after the first step and before the second, and leave the file in a bad state as a result. This was originally intended to be a rarely used process for this reason.

Reply:
I don't think we will go down this code path with any more frequency unless I'm missing something. This code should work identically as before. This was just meant to be a refactor to simplify this code path and remove the additional variable and branch.

Reply:
This change definitely alters the observed behavior in significant ways.

Reply:
mike i think it is the same.

Reply:
The 2-step rename is required when the ONLY change to the name is that the case of the base name changes, because onedrive will raise an exception like FileExists or similar. This is the only condition where it is absolutely required, and it is dangerous to use it for any other rename. If the code does not implement that use-case, then it must be fixed, otherwise it must be left as is.

Reply:
Ah... maybe the same. Will read again

Reply:
OK, I see it now, thanks for helping me get clarity on that. Looking at the code so "carefully" :-), and reiterating the intention does serve one possible purpose, however. Does this do a two step rename if the file moves to another folder, and also changes case simultaneously? It looks like it might, and it may not be necessary, and so could also be an issue.

Reply:
Perhaps:
`need_temp = (info.name.lower() == base.lower()) and (old_parent_id == new_parent_info.id)`

Reply:
ya I think you're right. I would like to have this in a separate PR though. I'm not sure if this needs to be in the hotfix

Reply:
That's fine. Please mention each time a PR will be a hotfix in the PR comments or PR description, that will help understand the context when reviewing, thx